src/type/validate.js

```diff
@@ -410,19 +425,41 @@ function validateObjectImplementsInterface(
     }
 
     // Assert additional arguments must not be required.
-    for (const objectArg of objectField.args) {
-      const argName = objectArg.name;
-      const ifaceArg = find(ifaceField.args, arg => arg.name === argName);
-      if (!ifaceArg && isRequiredArgument(objectArg)) {
+    for (const implementingArg of implementingField.args) {
+      const argName = implementingArg.name;
+      const implementedArg = find(
+        implementedField.args,
+        arg => arg.name === argName,
+      );
+      if (!implementedArg && isRequiredArgument(implementingArg)) {
         context.reportError(
-          `Object field ${object.name}.${fieldName} includes required argument ${argName} that is missing from the Interface field ${iface.name}.${fieldName}.`,
-          [objectArg.astNode, ifaceField.astNode],
+          `Object field ${type.name}.${fieldName} includes required argument ${argName} that is missing from the Interface field ${iface.name}.${fieldName}.`,
+          [implementingArg.astNode, implementedField.astNode],
         );
       }
     }
   }
 }
 
+function validateTypeImplementsAncestors(
+  context: SchemaValidationContext,
+  type: GraphQLObjectType | GraphQLInterfaceType,
+  iface: GraphQLInterfaceType,
+): void {
+  const implementingInterfaces = type.getInterfaces();
+  for (const transitive of iface.getInterfaces()) {
+    if (implementingInterfaces.indexOf(transitive) === -1) {
```

Comment:
This check doesn't for self-referencing interfaces:
```
interfaces Foo implements Foo {}
```
Fixed in https://github.com/mike-marcacci/graphql-js/pull/1

Reply:
Great catch!